### PR TITLE
Replace `CHECK_RETURN_IF_FALSE` with `RET_CHECK`

### LINKED ISSUE #579: Replace `CHECK_RETURN_IF_FALSE` with `RET_CHECK`
`CHECK_RETURN_IF_FALSE` is either a Stratum homegrown thing or really old. `RET_CHECK` replaces it with the same API.

Potential sources are 
https://github.com/google/xls/blob/c41a2732b3a6918edd497fd118156d1214e5fb57/xls/common/status/ret_check.h
or
https://github.com/google/aistreams/blob/09692b0a1940da590ffede933519fc41ec3405d9/aistreams/util/status/ret_check.h

#### GH wide code search
[RET_CHECK on GitHub](https://github.com/search?l=C%2B%2B&q=org%3Agoogle+RET_CHECK+extension%3Ah+extension%3Acc&type=Code)

[CHECK_RETURN_IF_FALSE on GitHub](https://github.com/search?q=org%3Agoogle+CHECK_RETURN_IF_FALSE+extension%3Ah+extension%3Acc&type=Code)

Also `xabsl`: https://github.com/google/xls/blob/c41a2732b3a6918edd497fd118156d1214e5fb57/xls/common/status/status_builder.h#L30-L33

### PR DESCRIPTION
**Pull request by pudelkoM**
This PR replaces the home-grown `CHECK_RETURN_IF_FALSE` macro with the more common `RET_CHECK` one. Semantics and behavior stay the same, no user-code changes needed. Code readability is improved a little, as the new macro is shorter, leading to less line splits.
We hope that this will make migration to `absl::Status` easier eventually, as we will probably go with whatever solution Google comes up with.

Fixes #579

### CODE REVIEW

File: stratum/lib/macros.h [hunk 1]
```diff
@@ -16,10 +16,12 @@ namespace stratum {
...
-#define CHECK_RETURN_IF_FALSE(cond) \
-  if (ABSL_PREDICT_TRUE(cond)) {    \
-  } else /* NOLINT */               \
-    return MAKE_ERROR(ERR_INVALID_PARAM) << "'" << #cond << "' is false. "
+
+// #define CHECK_RETURN_IF_FALSE(cond) \
+//   if (ABSL_PREDICT_TRUE(cond)) {    \
+//   } else /* NOLINT */               \
+//     return MAKE_ERROR(ERR_INVALID_PARAM) << "'" << #cond << "' is false. "
+#define CHECK_RETURN_IF_FALSE RET_CHECK
```
Comment (pudelkoM) [inline stratum/lib/macros.h:24]: WIP to see if code compiles and tests pass

**Review by Yi-Tseng — DISMISSED** [review]

File: stratum/hal/lib/bcm/bcm_table_manager.cc [hunk 2]
```diff
@@ -475,7 +474,7 @@ ::util::Status BcmTableManager::CommonFlowEntryToBcmFlowEntry(
   switch (common_flow_entry.action().type()) {
     case P4_ACTION_TYPE_PROFILE_MEMBER_ID: {
       uint32 member_id = common_flow_entry.action().profile_member_id();
-      ASSIGN_OR_RETURN(BcmNonMultipathNexthopInfo* member_nexthop_info,
+      ASSIGN_OR_RETURN(BcmNonMultipathNexthopInfo * member_nexthop_info,
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/bcm_table_manager.cc:477]: manually fix formatting, weird edge case

File: stratum/hal/lib/bcm/bcm_table_manager_test.cc [hunk 3]
```diff
@@ -431,15 +422,17 @@ const std::vector<std::pair<MappedField, BcmField>>& P4ToBcmFields() {
       type: P4_FIELD_TYPE_ETH_SRC
       value { u64: 11111111111111 }
       mask { u64: 99999999999999 }
-    )PROTO", &p4_field));
+    )PROTO",
+                                   &p4_field));
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/bcm_table_manager_test.cc:426]: manually fix formatting, weird edge case

File: stratum/hal/lib/bcm/sdklt/bcm_sdk_wrapper.cc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2018 Google LLC
 // Copyright 2018-present Open Networking Foundation
-// Copyright 2019 Broadcom
-// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019 Broadcom. All rights reserved. The term "Broadcom" refers to
+// Broadcom Inc. and/or its subsidiaries. SPDX-License-Identifier: Apache-2.0
```
Comment (pudelkoM) [inline stratum/hal/lib/bcm/sdklt/bcm_sdk_wrapper.cc:4]: fix

File: stratum/hal/lib/p4/p4_static_entry_mapper.cc [hunk 5]
```diff
@@ -2,7 +2,6 @@
 // Copyright 2018-present Open Networking Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-
```
Comment (pudelkoM) [inline stratum/hal/lib/p4/p4_static_entry_mapper.cc:5]: format in other PR

File: stratum/hal/lib/phal/fixed_layout_datasource.cc [hunk 6]
```diff
@@ -2,7 +2,6 @@
 // Copyright 2018-present Open Networking Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/fixed_layout_datasource.cc:5]: format in other PR

File: stratum/hal/lib/phal/phal_sim.cc [hunk 7]
```diff
@@ -2,13 +2,12 @@
 // Copyright 2018-present Open Networking Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/phal_sim.cc:5]: format in other PR

File: stratum/hal/lib/phal/system_fake.cc [hunk 8]
```diff
@@ -2,12 +2,11 @@
 // Copyright 2018-present Open Networking Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/system_fake.cc:5]: format in other PR

File: stratum/hal/lib/phal/system_real.cc [hunk 9]
```diff
@@ -2,14 +2,13 @@
 // Copyright 2018-present Open Networking Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/system_real.cc:5]: format in other PR

File: stratum/hal/lib/phal/attribute_group.cc [hunk 10]
```diff
@@ -285,25 +284,25 @@ class AttributeGroupInternal : public AttributeGroup,
 // macro can *only* be used in AddAttribute, since it pulls in variables
 // not explicitly declared as macro parameters (all of the variables captured by
 // the lambda). See usage below for context.
-#define ATTRIBUTE_SETTER_FUNCTION(proto_setter_function, type)               \
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/attribute_group.cc:288]: format in other PR

File: stratum/hal/bin/bcm/sim/main.cc [hunk 11]
```diff
@@ -2,10 +2,11 @@
 // Copyright 2018-present Open Networking Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-
```
Comment (pudelkoM) [inline stratum/hal/bin/bcm/sim/main.cc:5]: format in other PR

**Comment by pudelkoM** [comment]
Lint failure is expected, as `bfrt_table_manager_test.cc` and `bfrt_table_manager.cc` are unformatted.

**Pull request merged**